src/Database/PostgreSQL/PQTypes/Checks.hs

```diff
@@ -792,6 +792,22 @@ checkDBConsistency options domains tablesWithVersions migrations = do
           runSQL_ "COMMIT"
           runQuery_ (sqlCreateIndexConcurrently tname idx) `finally` begin
           updateTableVersion
+
+        ModifyColumnMigration cursorSql updateSql batchSize -> do
+          logMigration
+          ts <- getTransactionSettings
+          setTransactionSettings $ ts { tsIsolationLevel = Serializable }
```

Comment:
This alone is not sufficient. There also needs to be a `tsRestertPredicate` set that restarts the transaction when a db exception with a `SerializationFailure` error code is received.

Reply:
I'm not actually sure if we need serializable here. @marco44 what do you think?

The deal is that we want to modify a column by:

1. Adding a new column.
2. Adding a trigger that will copy values (with appropriate conversion) from the old column to the new column on insert/update.

Then, in a separate migration:

1. Copy data (with appropriate conversion) for existing rows from the old column to the new one, in batches.

And the question is, should the data be copied in serializable transactions or is the default isolation level sufficient?

Reply:
Serializable only works when all transactions working on the same data are set as serializable. If you really want to be sure no one touches the records while you're reading them, select for update is the way to go I think: you'll end up always reading "the latest version" of a record, and preventing anyone from touching them while you haven't committed. So I don't think it's what you're hoping for. For more details : https://wiki.postgresql.org/wiki/SSI (or just ask me)

But here, i'm not sure I understand the algorithm. We create a cursor using cursorSql, which is a query to get the PKs of all candidate records, right ?

If yes, then we just don't care about isolation: the cursor, when you start it, will get a "snapshot" of the DB like it is at the beginning of the query, what you're going to fetch is all records that match this query at the precise moment you run this query. So if we have new records in the meantime, those are going to be managed by the trigger. The only corner case I see, is if an update changes a PK value while we fix the records.